### PR TITLE
fix: cleanup autocomplete after milestone 3

### PR DESCRIPTION
**Pull request by callumcurtis**

### CODE REVIEW

**Review by camrohwer — APPROVED** [review]

File: packages/api/src/router/review.ts [hunk 1]
```diff
@@ -87,14 +103,29 @@ export const reviewRouter = createTRPCRouter({
...
+          createdAt: r.review.createdAt,
+          updatedAt: r.review.updatedAt,
+          userId: r.profile.id,
+          username: r.profile.username,
+          restaurantId: r.restaurant.id,
+          restaurantName: r.restaurant.name,
+          restaurantAddress: r.restaurant.address,
+        }),
+      );
       const hasMore = reviews.length > input.limit;
```
Comment (anormananderson) [inline packages/api/src/router/review.ts:129]: Is there any difference in how `const reviews` is defined here vs. in 'all' above?

File: packages/schema/src/atoms/index.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  .min(2, {
+    message: "Username must be at least 2 characters long",
+  })
+  .max(32, {
+    message: "Username must be at most 32 characters long",
+  })
+  .regex(/^[a-z0-9_.]{2,32}$/, {
+    message:
+      "Username must only contain lowercase letters, numbers, underscores, and periods",
+  });
```
Comment (anormananderson) [inline packages/schema/src/atoms/index.ts:14]: This allows usernames like '..' -- as long as that's intentional

**Review by anormananderson — APPROVED** [review]

File: packages/api/src/router/review.ts (hunk 1 above)
Comment (callumcurtis) [reply packages/api/src/router/review.ts:129]: Yeah, the `all` is getting reviews from all friends so the joins look different. They could be refactored to remove some duplication, but the fact they're similar is more of a coincidence as they could fork in the future as requirements change.

File: packages/schema/src/atoms/index.ts (hunk 2 above)
Comment (callumcurtis) [reply packages/schema/src/atoms/index.ts:14]: We could come up with some more restrictive constraints but I kept it simple here for now. For reference, Discord disallows having multiple periods in a row - we could do the same for special characters (here, underscores and periods).

**Review by anormananderson — APPROVED** [review]
Nice find with those Lucide components.

File: packages/schema/src/atoms/index.ts (hunk 2 above)
Comment (anormananderson) [reply packages/schema/src/atoms/index.ts:14]: Something like /^(?:[_.]?[a-z0-9][_.]?)+$/ if we want to match Discord? The validation of the actual length should already be taken care of with the validation above the regex

Comment (callumcurtis) [reply packages/schema/src/atoms/index.ts:14]: I like it, I'll double check that regex and update my PR stack

Comment (callumcurtis) [reply packages/schema/src/atoms/index.ts:14]: I'll make a separate PR for this

**Pull request merged**